### ISSUE TITLE
afmt OOM 

### ISSUE THREAD

**Issue by zonkzonk** [radare/radare2#5584]
morrn,

see attached valgrind.

how to reproduce:

```
r2 /home/zl/b.mp3
[0x0000b420]> s 0
[0x00000000]> S
[0x00000000]> afmt
Cannot find function
[0x00000000]> af
af        af        afa       afan      afc       
afi       afb       afbb      afn       afr       
afs       af*       afv       afvn      
[0x00000000]> af
af        af        afa       afan      afc       
afi       afb       afbb      afn       afr       
afs       af*       afv       afvn      
[0x00000000]> aftm
cant find function here
[0x00000000]> af
[0x00000000]> pdf
/ (fcn) fcn.00000000 14
|           0x00000000      fff3           push rbx
|           0x00000002      d0c0           rol al, 1
|           0x00000004      0000           add byte [rax], al
|           0x00000006      0000           add byte [rax], al
|           0x00000008      0000           add byte [rax], al
|           0x0000000a      0000           add byte [rax], al
\           0x0000000c      00d0           add al, dl
[0x00000000]> afmt
Merge 0x00000000 into 0x00000000
Killed


 181.572047] systemd invoked oom-killer: gfp_mask=0x201da, order=0, oom_score_adj=0
[  181.572049] systemd cpuset=/ mems_allowed=0
[  181.572054] CPU: 0 PID: 1 Comm: systemd Not tainted 4.2.0-42-generic #49-Ubuntu
[  181.572055] Hardware name: LENOVO 20AUS00N00/20AUS00N00, BIOS J4ET34WW(1.07) 09/06/2013
[  181.572056]  0000000000000286 00000000ee73da3e ffff88011a74f978 ffffffff817f318e
[  181.572059]  ffff88011a7f0000 ffff88011a7f06c0 ffff88011a74fa08 ffffffff817f0f49
[  181.572060]  ffff88011a74f9e8 ffffffffc0152611 ffff88011a74f9a8 0000000000000000
[  181.572062] Call Trace:
[  181.572068]  [<ffffffff817f318e>] dump_stack+0x63/0x81
[  181.572071]  [<ffffffff817f0f49>] dump_header+0x8c/0x214
[  181.572102]  [<ffffffffc0152611>] ? i915_gem_shrinker_oom+0x1a1/0x200 [i915]
[  181.572106]  [<ffffffff811881e1>] oom_kill_process+0x1e1/0x3b0
[  181.572108]  [<ffffffff81188966>] out_of_memory+0x556/0x5b0
[  181.572111]  [<ffffffff8118e6d5>] __alloc_pages_nodemask+0x8e5/0xa30
[  181.572113]  [<ffffffff81399b47>] ? __blk_run_queue+0x37/0x50
[  181.572116]  [<ffffffff811d6101>] alloc_pages_current+0x91/0x100
[  181.572118]  [<ffffffff81184700>] __page_cache_alloc+0xa0/0xc0
[  181.572120]  [<ffffffff81186c8a>] filemap_fault+0x14a/0x3f0
[  181.572123]  [<ffffffff811b32a0>] __do_fault+0x50/0xe0
[  181.572125]  [<ffffffff811b8306>] handle_mm_fault+0xf96/0x1800
[  181.572128]  [<ffffffff81068ab7>] __do_page_fault+0x197/0x400
[  181.572130]  [<ffffffff81068d42>] do_page_fault+0x22/0x30
[  181.572133]  [<ffffffff817fbfc8>] page_fault+0x28/0x30
[  181.572134] Mem-Info:
[  181.572137] active_anon:687547 inactive_anon:213019 isolated_anon:0
                active_file:90 inactive_file:136 isolated_file:0
                unevictable:8 dirty:0 writeback:0 unstable:0
                slab_reclaimable:3545 slab_unreclaimable:4097
                mapped:2274 shmem:5594 pagetables:5818 bounce:0
```

valgrind:

![meh](https://cloud.githubusercontent.com/assets/5694980/17859838/81a16b54-688b-11e6-9d23-1591dbbfdb8d.png)

,r2 -v
radare2 0.10.6-git 12149 @ linux-x86-64 git.0.10.5-15-ge0ee8cd
commit: e0ee8cdefa96cda4e3d7880561ffb4d94fedddd7 build: 2016-08-19

**Comment by Maijin** [comment]
Where is the valgrind log?

**Comment by zonkzonk** [comment]
see above!

**Comment by oddcoder** [comment]
no logs ?

**Comment by radare** [comment]
just curl https://cloud.githubusercontent.com/assets/5694980/17859838/81a16b54-688b-11e6-9d23-1591dbbfdb8d.png

**Comment by zonkzonk** [comment]
@radare Thanks for pointing out. still not used to gists

**Comment by radare** [comment]
anyway, is afmt a commant at all?

**Comment by radare** [comment]
it should say: invalid command

**Comment by zonkzonk** [comment]
strange

**Comment by oddcoder** [comment]
@radare aftm is type matching

**Comment by radare** [comment]
yes aftm it is, but afmt is not

**Comment by oddcoder** [comment]
can you share that mp3 file

**Comment by oddcoder** [comment]
@zonkzonk can you share the file caused that bug ?

**Comment by zonkzonk** [comment]
ofc: 

![b](https://cloud.githubusercontent.com/assets/5694980/17882328/2994125a-690c-11e6-8273-a278eb04aca0.png)

**Comment by oddcoder** [comment]
fixed no longer crashes

**Comment by zonkzonk** [comment]
still works with manual seeking:

`[0x00000000]> af
[0x00000000]> pdf
/ (fcn) fcn.00000000 14
|           0x00000000      fff3           push rbx
|           0x00000002      d0c0           rol al, 1
|           0x00000004      0000           add byte [rax], al
|           0x00000006      0000           add byte [rax], al
|           0x00000008      0000           add byte [rax], al
|           0x0000000a      0000           add byte [rax], al
\           0x0000000c      00d0           add al, dl
[0x00000000]> s b420
[0x00000000]> s 0xb420
[0x0000b420]> s 0
[0x00000000]> S
[0x00000000]> afmt
Merge 0x00000000 into 0x00000000
Killed

```

**Comment by radare** [comment]
afm was not in help but it is a function merge command

**Comment by radare** [comment]
this crash can be reproduced with :

af;afm $$

**Comment by oddcoder** [comment]
lol, no wonders why I couldn't reproduce :P

**Comment by radare** [comment]
Fixed in a3ce21645e60387d43a30ba4634c0f627e26b373
